android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -1888,9 +1888,12 @@ public static boolean updateText(KeyboardType kbType, String text) {
 
   public static boolean updateSelectionRange(KeyboardType kbType, int selStart, int selEnd) {
     boolean result = false;
+    // Currently, KMW doesn't support selection on touch alias elements
+    // so pass null context to KMW. See issue #5853
```

Comment:
Does this actually pass in null context, or just no selection? To test this, we need a keyboard with a context rule, for example:

```
'a' + 'b' > 'x'
```

Then, have the text `123a` selected, and type <kbd>b</kbd>. The selection should be deleted, and `b` emitted -- not `x`.

Reply:
Ah, it's just passing no selection, so KMW still has the context.

Testing with xinaliq keyboard and the selection isn't correct...

How should a null context be passed to KMW then?

Reply:
Yes, that should do it, I think.

Reply:
Maybe it's moot calling `updateKMSelectionRange()` afterwards.

The app seems to behave with only `updateKMText()`